src/promotion/ops.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright(c) 2019 Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause-Clear
+ */
+
+#ifndef PROMOTION_OPS_H_
+#define PROMOTION_OPS_H_
+
+#include "../metadata/metadata.h"
+#include "promotion.h"
+
+struct ocf_promotion_policy {
+	ocf_promotion_t type;
+	void *ctx;
+};
+
+struct promotion_policy_ops {
+	const char *name;
+		/*!< Promotion policy name */
+
+	ocf_error_t (*init)(ocf_cache_t cache, ocf_promotion_policy_t policy);
+		/*!< Allocate and initialize promotion policy */
+
+	void (*deinit)(ocf_promotion_policy_t policy);
+		/*!< Deinit and free promotion policy */
+
+	ocf_error_t (*set_param)(ocf_promotion_policy_t policy, uint8_t param_id,
```

Comment:
pls add get_param

Reply:
Will do in the next PR together with nhit policy implementation!